assets/app/scripts/directives/nav.js

```diff
@@ -32,7 +32,7 @@ angular.module('openshiftConsole')
             $('<option>')
               .attr("value", project.metadata.name)
               .attr("selected", project.metadata.name == $scope.projectName)
-              .text(project.displayName || project.metadata.name)
+              .text(project | displayName || project.metadata.name)
```

Comment:
the filter syntax ( | ) only works from HTML templates.  in there you are just in javascript so you have to do project.metadata.annotation.displayName, keeping in mind that annotation may not exist so its (project.metadata.annotations && project.metadata.annotations.displayName)